decentraland/screens/MyAccountScreen.js

```diff
@@ -1,53 +1,73 @@
+import { MISSING, DONE } from "@constants/ActionStatus";
 import PropTypes from "prop-types";
 
+const StepsWithAction = [
+  {
+    creationStatus: FUNDING_WITH_ETH,
+    name: "Funded with ETH",
+    percentage: 0.25,
+  },
+  {
+    creationStatus: FUNDING_WITH_MANA,
+    name: "Funded with MANA tokens",
+    percentage: 0.25,
+  },
+  {
+    creationStatus: APPROVING_MARKETPLACE,
+    name: "Linked to marketplace",
+    percentage: 0.25,
+  },
+];
+
 export class MyAccountScreen extends React.Component {
   render() {
     const { accountInfo } = this.props;
-    const creationActions = [];
-    creationActions.push({
+    const { creationActions, account } = accountInfo;
+    const isAccountCreated = account !== null;
+
+    const creationSteps = [];
+
+    creationSteps.push({
       name: "Account created",
-      action: null,
-      status: !accountInfo.account ? ActionStatus.MISSING : ActionStatus.DONE,
+      creationStatus: null,
+      status: isAccountCreated ? DONE : MISSING,
+      percentage: 0.25,
     });
-    Object.keys(AccountCreationActions).forEach(action => {
-      const creationAction = {
-        name: AccountCreationActions[action].name,
-        action,
-      };
+
+    StepsWithAction.forEach(stepWithAction => {
+      const { name, creationStatus, percentage } = stepWithAction;
+
       // TODO: As soon as we store action status in redux, this logic will change
       // transaction pending, confirmed once, confirmed many times, failed, etc.
-      if (accountInfo.creationActions.hasOwnProperty(action)) {
-        creationAction.status = ActionStatus.DONE;
-      } else {
-        creationAction.status = ActionStatus.MISSING;
-      }
-      creationActions.push(creationAction);
+      const status = creationActions[creationStatus] ? DONE : MISSING;
+
+      const creationStep = { name, creationStatus, status, percentage };
+
+      creationSteps.push(creationStep);
     });
-    const isAccountCreated = !!accountInfo.account;
+
     return (
       <MyAccount
-        progress={this._getPercentage(isAccountCreated, creationActions)}
-        creationActions={creationActions}
+        progress={this._getPercentage(creationSteps)}
+        creationSteps={creationSteps}
       />
     );
   }
 
-  _getPercentage(isAccountCreated, accountActions) {
-    let percentage = isAccountCreated ? 0.25 : 0;
-    accountActions.forEach(accountAction => {
-      const isDone = accountAction.status === ActionStatus.DONE;
-      const hasDefinedActionPercentage = AccountCreationActions.hasOwnProperty(
-        accountAction.action
-      );
-      if (isDone && hasDefinedActionPercentage) {
-        percentage += AccountCreationActions[accountAction.action].percentage;
-      }
-    });
+  _getPercentage(creationSteps) {
+    const percentage = creationSteps
+      .filter(step => step.status === DONE)
+      .map(step => step.percentage)
+      .reduce((total, stepPercentage) => total + stepPercentage, 0);
```

Comment:
I haven't used reduce much yet. It's interesting that it has access to `total` in this way.

Reply:
Yes, neither do I, seems a good way to do accumulations.

Reply:
Yep, cool. Resolving this.